molecule/default/tests/test_default.py

```diff
@@ -33,38 +34,30 @@ def test_pip3_packages(host, pkg):
 
 
 @pytest.mark.parametrize(
-    "file,content",
+    "prop,regex",
     [
+        ("After", r"^After=.*network\.target"),
+        ("After", r"^After=.*multi-user\.target"),
+        ("After", r"^After=.*cloud-final\.service"),
         (
-            "/lib/systemd/system/mongod.service",
-            r"^After=network.target multi-user.target cloud-final.service$",
+            "ExecStart",
+            r"^ExecStart=.*argv\[\]=/usr/bin/numactl --interleave=all /usr/bin/mongod --config /etc/mongod\.conf",
         ),
-        (
-            "/lib/systemd/system/mongod.service",
-            r"^ExecStart=/usr/bin/numactl --interleave=all /usr/bin/mongod --config /etc/mongod.conf$",
-        ),
-        (
-            "/lib/systemd/system/mongod.service",
-            r"^RequiresMountsFor=/var/lib/mongodb /var/lib/mongodb/journal /var/log/mongodb$",
-        ),
-        (
-            "/lib/systemd/system/mongod.service",
-            r"^AssertPathIsMountPoint=/var/lib/mongodb$",
-        ),
-        (
-            "/lib/systemd/system/mongod.service",
-            r"^AssertPathIsMountPoint=/var/log/mongodb$",
-        ),
-        ("/lib/systemd/system/mongod.service", r"^RuntimeDirectory=mongodb$"),
-        ("/lib/systemd/system/mongod.service", r"^RuntimeDirectoryMode=0744$"),
+        ("RequiresMountsFor", r"^RequiresMountsFor=.*/var/lib/mongodb"),
+        ("RequiresMountsFor", r"^RequiresMountsFor=.*/var/lib/mongodb/journal"),
+        ("RequiresMountsFor", r"^RequiresMountsFor=.*/var/log/mongodb"),
+        ("RuntimeDirectory", r"^RuntimeDirectory=mongodb$"),
+        ("RuntimeDirectoryMode", r"^RuntimeDirectoryMode=0744$"),
     ],
 )
-def test_files_content(host, file, content):
-    """Test that config files were modified as expected."""
-    f = host.file(file)
-
-    assert f.exists
-    assert f.contains(content)
+def test_unit_content(host, prop, regex):
```

Comment:
Since we're using a drop-in file I don't think this test is very useful now. I think testing that the drop-in directory and file exist and have the appropriate permissions is more inline with what the role does as of this PR.

Reply:
I disagree.  I think there is some value in verifying that the values we intend to set actually get set after `systemd`'s munging of unit files and drop-ins.  The presence of this function actually caught a bug when I was working on this branch; I was accidentally writing the drop-in to the wrong location and it wasn't being picked up.

That said, if I remove this function then I can close #33.  I will defer to the will of @cisagov/vm-dev.

Reply:
Ah, I was in a rush and didn't notice the way the function was checking contents changed. I retract my statement about removing this test. However, I think we should _also_ test that the directory and file being created in the role both exist and have the correct permissions.

Reply:
Please see commits 41a4dcd and f4f2579.